pkg/ovncontroller/network.go

```diff
@@ -52,23 +57,36 @@ func CreateAdditionalNetworks(
 					physNet, interfaceName, err)
 			}
 
+			ownerRef := metav1.NewControllerRef(instance, instance.GroupVersionKind())
 			nad = &netattdefv1.NetworkAttachmentDefinition{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      physNet,
-					Namespace: instance.Namespace,
-					Labels:    labels,
-				},
-				Spec: netattdefv1.NetworkAttachmentDefinitionSpec{
-					Config: fmt.Sprintf(
-						`{"cniVersion": "0.3.1", "name": "%s", "type": "host-device", "device": "%s"}`,
-						physNet, interfaceName),
+					Name:            physNet,
+					Namespace:       instance.Namespace,
+					Labels:          labels,
+					OwnerReferences: []metav1.OwnerReference{*ownerRef},
 				},
+				Spec: nadSpec,
 			}
 			// Request object not found, lets create it
 			if err := h.GetClient().Create(ctx, nad); err != nil {
 				return nil, fmt.Errorf("can not create NetworkAttachmentDefinition %s/%s: %w",
 					physNet, interfaceName, err)
 			}
+		} else {
+			owned := false
+			for _, owner := range nad.GetOwnerReferences() {
+				if owner.Name == instance.Name {
+					owned = true
+					break
+				}
+			}
+			if owned {
+				nad.Spec = nadSpec
+				if err := h.GetClient().Update(ctx, nad); err != nil {
+					return nil, fmt.Errorf("can not update NetworkAttachmentDefinition %s/%s: %w",
```

Comment:
nit: can not -> cannot

```
Don’t use can not when you mean cannot. The only time you’re likely to see can not written as separate words is when the word “can” happens to precede some other phrase that happens to start with “not”:
```

(though I see it's the same above for create case... so I dunno...)

Reply:
Update at all places